lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -277,7 +298,15 @@ def _patch(self, _) -> None:
             if self._is_patched(client):
                 return
             if self.service_name != self._app:
-                self._delete_and_create_service(client)
+                if not self.service_type == "LoadBalancer":
+                    self._delete_and_create_service(client)
+                else:
+                    self._delete_and_create_lb_service(client)
+            elif self.service_type == "LoadBalancer":
+                logger.error(
```

Comment:
Should we `raise ValueError` here instead?

Reply:
removed altogether in favor of pre-setting the service name